tests/timelord/test_new_peak.py

```diff
@@ -43,103 +46,344 @@ async def test_timelord_new_peak_basic(
 
                 blocks_1 = bt.get_consecutive_blocks(2, blocks)
                 await _validate_and_add_block(b1, blocks_1[-2])
                 await _validate_and_add_block(b1, blocks_1[-1])
-                await timelord_api.new_peak_timelord(timelord_peak_from_block(blocks_1[-2], b1, bt.constants))
-                await timelord_api.new_peak_timelord(timelord_peak_from_block(blocks_1[-1], b1, bt.constants))
+                await timelord_api.new_peak_timelord(timelord_peak_from_block(b1, blocks_1[-2]))
+                await timelord_api.new_peak_timelord(timelord_peak_from_block(b1, blocks_1[-1]))
                 assert timelord_api.timelord.new_peak.reward_chain_block.height == blocks_1[-1].height
 
-                # # new unknown peak, weight less then curr peak
-                # blocks_2 = bt.get_consecutive_blocks(1, blocks)
-                # await _validate_and_add_block(b2, blocks_2[-1])
-                # await timelord_api.new_peak_timelord(timelord_peak_from_block(blocks_2[-1], b2, bt.constants))
-                # assert timelord_api.timelord.new_peak.reward_chain_block.height == blocks_1[-1].height
+                # new unknown peak, weight less then curr peak
+                blocks_2 = bt.get_consecutive_blocks(1, blocks)
+                await _validate_and_add_block(b2, blocks_2[-1])
+                await timelord_api.new_peak_timelord(timelord_peak_from_block(b2, blocks_2[-1]))
+                assert timelord_api.timelord.last_state.last_weight == blocks_1[-1].weight
+                assert timelord_api.timelord.last_state.total_iters == blocks_1[-1].reward_chain_block.total_iters
 
     @pytest.mark.anyio
-    async def test_timelord_new_peak_heavier_unfinished(
+    async def test_timelord_new_peak_unfinished_not_orphaned(
         self, bt: BlockTools, timelord: Tuple[TimelordAPI, ChiaServer], default_1000_blocks: List[FullBlock]
     ) -> None:
+        async with create_blockchain(bt.constants, 2) as (b1, db_wrapper1):
+            timelord_api, _ = timelord
+            for block in default_1000_blocks:
+                await _validate_and_add_block(b1, block)
+
+            peak = timelord_peak_from_block(b1, default_1000_blocks[-1])
+            assert peak is not None
+            assert timelord_api.timelord.new_peak is None
+            await timelord_api.new_peak_timelord(peak)
+            assert timelord_api.timelord.new_peak is not None
+            assert timelord_api.timelord.new_peak.reward_chain_block.get_hash() == peak.reward_chain_block.get_hash()
+
+            # make two new blocks on tip
+            blocks_1 = bt.get_consecutive_blocks(2, default_1000_blocks)
+            block_1 = blocks_1[-2]
+            block_2 = blocks_1[-1]
+            await _validate_and_add_block(b1, block_1)
+            await _validate_and_add_block(b1, block_2)
+
+            block_record = b1.block_record(block_2.header_hash)
+            sub_slot_iters, difficulty = get_next_sub_slot_iters_and_difficulty(
```

Comment:
I'd merge `get_next_sub_slot_iters_and_difficulty` and `timelord_protocol.NewUnfinishedBlockTimelord` into a single function (for example `get_unfinished_block_timelord`), taking the inputs `bt`, `b1`, `block_record` and `block_1` and returning `timelord_protocol.NewUnfinishedBlockTimelord`. I think this could simplify the test code, since we are repeating these complicated calculations 4 times across all tests.

Reply:
get_next_sub_slot_iters_and_difficulty is not really related to NewUnfinishedBlockTimelord and is used all over the full_node without calling NewUnfinishedBlockTimelord so that wont make much sense